DigitalLearningSolutions.Web/Scripts/trackingSystem/emailDelegates.ts

```diff
@@ -10,10 +10,17 @@ new SearchSortFilterAndPaginate(route, false, false, true, 'EmailDelegateFilter'
 setUpSelectAndDeselectButtons();
 
 function alertResultCount(): void {
-  // un-assign and re-assign the alert role to results-count so a screen-reader re-reads it
+  // change the content of the search results
+  // so that an "identical" result is announced by the aria-live attribute
   const resultCount = document.getElementById('results-count') as HTMLSpanElement;
-  resultCount.setAttribute('role', '');
-  resultCount.setAttribute('role', 'alert');
+  const resultCountMessage = resultCount.innerHTML;
+  const indexOfSpace = resultCountMessage.search('&nbsp');
+
+  if (indexOfSpace === -1) {
+    resultCount.innerHTML = `${resultCountMessage}&nbsp`;
+  } else {
+    resultCount.innerHTML = resultCountMessage.substring(0, indexOfSpace);
+  }

```

Comment:
Removing and reassigning the alert role seemed to be inconsistently picked up by browsers, if at all. As the alert role (and the potential new replacement `aria-live `) both announce any changes to the content, the reliable way to get them to announce the number of results would be changing the content. 

In this case, we are simply selecting check boxes, so there are no changes to the number of results. In order to force the button press to announce the number of results (so that it is obvious the button has actually done something), we add a no-break space to the end of the message (or remove it if it already exists) so that there is a clear change in the message so it is read out.

Reply:
As with the other one, can we put the new stuff in a function of its own?